### PR TITLE
Important fix for race mode

### PR DESCRIPTION
**Pull request by TheAxeMan301**
Someone finally figured out that race mode was not actually altering the seed! This fix makes it so that race mode really does give a different seed. Downside is that past weeklies will no longer roll properly! Though it will work if you uncheck race mode.

### CODE REVIEW

File: Randomizer.cpp [hunk 1]
```diff
@@ -844,9 +844,9 @@ namespace Randomizer {
         */
         if (options.race) {
             std::cout << "Race mode enabled, altering actual seed.\n";
-            Random::RandomInit(seed);
-            Random::RandomInit(seed);
-            Random::RandomInit(seed);
+            Random::RandomInteger(100);
```
Comment (hellow554) [inline Randomizer.cpp]: You sure about the 100 here? Shouldn't you use `seed` here?

Comment (TheAxeMan301) [reply Randomizer.cpp]: It actually doesn't matter, we just need to do something that advances the RNG. Maybe I should use a different function entirely.

Comment (TheAxeMan301) [reply Randomizer.cpp]: Hey there actually is a function to do exactly this! Thanks for the reviews, it is really helping with code quality here.

**Comment by hellow554** [comment]
LGTM

File: Random.h [hunk 2]
```diff
@@ -9,6 +9,8 @@ namespace Random {
     int RandomInteger(int Range);
     // Returns [LB, UB]
     int RandomIntegerRange(int LowerBound, int UpperBound);
+    // Advance the RNG state, by "throwing away" some numbers
+    void discard(unsigned int z);
```
Comment (hellow554) [inline Random.h]: Nitpick: the standard says `unsigned long long`

https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine/discard

**Pull request merged**